### PR TITLE
chore: cleanup ETH/FOX base LP/staking opportunities

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

See https://github.com/shapeshift/web/pull/3760#discussion_r1099699436 - lots of base opportunities we were using only for the purpose of getting the name (which is already existing on the opportunity base) and rewardAddress (the same for all ETH/FOX opportunities).

Does an extensive types and constants cleanup, with more accurate types while at it.

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

## Testing

- DeFi opportunities are still happy
- FOX Farming LP/Staking opportunities are still happy, both in the DeFi section, in the FOX Page, in /accounts, account, and asset pages

### Engineering

- Refactor looks sound

### Operations

- ☝🏽 

## Screenshots (if applicable)

image
image
image
image
image
image
image

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3761** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3761" target="_blank">Graphite</a> 
    * **PR #3762** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3762" target="_blank">Graphite</a> 
      * **PR #3760** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3760" target="_blank">Graphite</a> 
        * **PR #3777** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3777" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/3777?utm_source=stack-comment).

File: src/plugins/foxPage/FoxCommon.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ export enum OpportunityTypes {
 }
 
 export type ExternalOpportunity = {
-  opportunityName: string | undefined
+  name: string | undefined
```
Comment (gomesalexandre) [inline src/plugins/foxPage/FoxCommon.ts:14]: Next up let's remove this altogether but better for now

File: src/plugins/foxPage/hooks/useOtherOpportunities.ts [hunk 2]
```diff
@@ -46,16 +42,6 @@ export const useOtherOpportunities = (assetId: AssetId) => {
     selectHighestBalanceAccountIdByLpId(state, highestBalanceLpAccountIdFilter),
   )
 
-  const farmingv5EarnOpportunityFilter = useMemo(
```
Comment (gomesalexandre) [inline src/plugins/foxPage/hooks/useOtherOpportunities.ts:49]: No idea why we were selecting the *earn* opportunity (metadata + user data parsed as `EarnOpportunityType`) when all we needed was metadata

File: src/pages/Defi/components/OpportunityCard.tsx [hunk 3]
```diff
@@ -132,7 +132,7 @@ export const OpportunityCard = ({
     }
   }, [asset, assetId, chainId, getVaultName, opportunityName, provider, version])
 
-  if (!asset) return null
+  if (!(asset && cryptoAmountPrecision)) return null
```
Comment (gomesalexandre) [inline src/pages/Defi/components/OpportunityCard.tsx:135]: revert
```suggestion
  if (!asset) return null
```

File: src/state/slices/opportunitiesSlice/selectors/stakingSelectors.ts [hunk 4]
```diff
@@ -285,19 +288,38 @@ export const selectAggregatedEarnUserStakingOpportunityByStakingId = createDeepE
...
-        .map(icon => icon ?? ''),
-      opportunityName: opportunity.name,
-    }),
+  (opportunity, marketData, assets): StakingEarnOpportunityType | undefined => {
+    if (!opportunity) return
+
+    const asset = assets[opportunity.assetId]
+    const underlyingAsset = assets[opportunity.underlyingAssetId]
+
+    const aggregatedEarnUserStakingOpportunity: StakingEarnOpportunityType = Object.assign(
```
Comment (gomesalexandre) [inline src/state/slices/opportunitiesSlice/selectors/stakingSelectors.ts:297]: Assigned to a const for type assignment purposes, nothing really changed here except for the addition of `cryptoAmountBaseUnit` and `cryptoAmountPrecision` for types completeness

File: src/state/slices/opportunitiesSlice/selectors/stakingSelectors.ts [hunk 5]
```diff
@@ -431,25 +459,27 @@ export const selectAggregatedEarnUserStakingOpportunitiesIncludeEmpty =
...
-              }
+
               return {}
             })(),
-            STAKING_EARN_OPPORTUNITIES[opportunity.assetId],
             opportunity,
             {
               chainId: fromAssetId(opportunity.assetId).chainId,
               cryptoAmountBaseUnit: '0',
+              cryptoAmountPrecision: '0',
```
Comment (gomesalexandre) [inline src/state/slices/opportunitiesSlice/selectors/stakingSelectors.ts:482]: Needed for types completeness now that we've removed dem casted consts

File: src/state/slices/opportunitiesSlice/selectors/stakingSelectors.ts [hunk 6]
```diff
@@ -525,11 +555,14 @@ export const selectEarnUserStakingOpportunityByUserStakingId = createDeepEqualOu
...
-    return {
-      ...LP_EARN_OPPORTUNITIES[userStakingOpportunity.assetId ?? ''],
+    const earnUserStakingOpportunity: StakingEarnOpportunityType = {
       ...userStakingOpportunity,
+      isLoaded: true,
       chainId: fromAssetId(userStakingOpportunity.assetId).chainId,
       cryptoAmountBaseUnit: userStakingOpportunity.stakedAmountCryptoBaseUnit ?? '0',
+      cryptoAmountPrecision: bnOrZero(userStakingOpportunity.stakedAmountCryptoBaseUnit)
+        .div(bn(10).pow(asset?.precision ?? underlyingAsset?.precision ?? 1))
+        .toFixed(),
```
Comment (gomesalexandre) [inline src/state/slices/opportunitiesSlice/selectors/stakingSelectors.ts:565]: Needed for types completeness now that we've removed casted consts

File: src/state/slices/opportunitiesSlice/types.ts [hunk 7]
```diff
@@ -38,7 +38,7 @@ export type OpportunityMetadataBase = {
   // TODO: Optional for backwards compatibility, but it should always be present
   rewardAssetIds?: AssetIdsTuple
   expired?: boolean
-  name?: string
+  name: string
```
Comment (gomesalexandre) [inline src/state/slices/opportunitiesSlice/types.ts:41]: This should never be undefined really

File: src/state/slices/opportunitiesSlice/opportunitiesSlice.test.ts [hunk 8]
```diff
@@ -38,6 +38,7 @@ describe('opportunitiesSlice', () => {
               // The LP token AssetId
               assetId: foxEthLpAssetId,
               id: foxEthLpAssetId,
+              name: 'ETH/FOX LP',
```
Comment (gomesalexandre) [inline src/state/slices/opportunitiesSlice/opportunitiesSlice.test.ts:41]:
```suggestion
              name: 'ETH/FOX Pool',
```

File: src/state/slices/opportunitiesSlice/opportunitiesSlice.test.ts [hunk 9]
```diff
@@ -64,6 +65,7 @@ describe('opportunitiesSlice', () => {
...
```
Comment (gomesalexandre) [inline src/state/slices/opportunitiesSlice/opportunitiesSlice.test.ts:68]:
```suggestion
              name: 'ETH/FOX Pool',
```

File: src/state/slices/opportunitiesSlice/opportunitiesSlice.test.ts [hunk 10]
```diff
@@ -88,6 +90,7 @@ describe('opportunitiesSlice', () => {
...
```
Comment (gomesalexandre) [inline src/state/slices/opportunitiesSlice/opportunitiesSlice.test.ts:93]:
```suggestion
              name: 'ETH/FOX Pool',
```

File: src/state/slices/opportunitiesSlice/selectors/stakingSelectors.ts [hunk 11]
```diff
@@ -440,8 +465,7 @@ export const selectActiveAggregatedEarnUserStakingOpportunitiesWithTotalFiatAmou
           }).toFixed(),
         })),
   )
-
-// Returns a single aggregated amount, for all opportunities, accounts, and assets
+// Used exclusively in useEarnBalances - returns a single aggregated amount, for all opportunities, accounts, and assets
```
Comment (0xdef1cafe) [inline src/state/slices/opportunitiesSlice/selectors/stakingSelectors.ts:468]: not any more it's not

File: src/state/slices/opportunitiesSlice/selectors/stakingSelectors.ts [hunk 12]
```diff
@@ -440,8 +465,7 @@ export const selectActiveAggregatedEarnUserStakingOpportunitiesWithTotalFiatAmou
...
 // Including delegations, undelegations, and rewards
 // Also slaps in ETH/FOX balances which value lives in the portfolio vs. being an "upstream earn opportunity"
 export const selectEarnBalancesFiatAmountFull = createDeepEqualOutputSelector(
```
Comment (0xdef1cafe) [inline src/state/slices/opportunitiesSlice/selectors/stakingSelectors.ts:471]: delet?

File: src/state/slices/opportunitiesSlice/selectors/stakingSelectors.ts (hunk 11 above)
Comment (gomesalexandre) [inline src/state/slices/opportunitiesSlice/selectors/stakingSelectors.ts:468]:
```suggestion
```

Comment (gomesalexandre) [inline src/state/slices/opportunitiesSlice/selectors/stakingSelectors.ts:468]:
```suggestion
// Returns a single aggregated amount, for all opportunities, accounts, and assets
```

**Review by 0xdef1cafe — APPROVED** [review]
stamped with comment

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**